Libraries/Cordio/wsf/sources/targets/baremetal/wsf_nvm.c

```diff
@@ -21,59 +21,88 @@
+#define WSF_NVM_START_ADDR 0x0000
 
 /*! Reserved filecode. */
-#define WSF_NVM_RESERVED_FILECODE                 ((uint64_t)0)
+#define WSF_NVM_RESERVED_FILECODE ((uint64_t)0)
 
 /* Unused (erased) filecode. */
 /* TODO: May depend on flash type */
-#define WSF_NVM_UNUSED_FILECODE                   ((uint64_t)0xFFFFFFFFFFFFFFFF)
+#define WSF_NVM_UNUSED_FILECODE ((uint64_t)0xFFFFFFFFFFFFFFFF)
 
 /*! Flash word size. */
-#define WSF_FLASH_WORD_SIZE   4
+#define WSF_FLASH_WORD_SIZE 4
 
 /*! Align value to word boundary. */
-#define WSF_NVM_WORD_ALIGN(x)                     (((x) + (WSF_FLASH_WORD_SIZE - 1)) & \
-                                                         ~(WSF_FLASH_WORD_SIZE - 1))
+#define WSF_NVM_WORD_ALIGN(x) (((x) + (WSF_FLASH_WORD_SIZE - 1)) & ~(WSF_FLASH_WORD_SIZE - 1))
 
-#define WSF_NVM_CRC_INIT_VALUE                    0xFEDCBA98
+#define WSF_NVM_CRC_INIT_VALUE 0xFEDCBA98
 
 /**************************************************************************************************
   Data Types
 **************************************************************************************************/
 
 /*! \brief      Header. */
-typedef struct
-{
-  uint64_t          id;         /*!< Stored data ID. */
-  uint32_t          len;        /*!< Stored data length. */
-  uint32_t          headerCrc;  /*!< CRC of this header. */
-  uint32_t          dataCrc;    /*!< CRC of subsequent data. */
+
+typedef struct {
+    uint64_t id; /*!< Stored data ID. */
+    uint32_t len; /*!< Stored data length. */
+    uint32_t headerCrc; /*!< CRC of this header. */
+    uint32_t dataCrc; /*!< CRC of subsequent data. */
 } WsfNvmHeader_t;
 
-static struct
-{
-  uint32_t          availAddr;  /*!< Next available address for NVM write. */
-  uint32_t          sectorSize; /*!< Size of erase sector. */
-  uint32_t          totalSize;  /*!< Total size of NVM storage. */
+#define WSF_NVM_HEADER_SIZE sizeof(WsfNvmHeader_t)
+#define WSF_NVM_FILE_SIZE(header_len) (WSF_NVM_HEADER_SIZE + (header_len))
+static struct {
+    uint32_t availAddr; /*!< Next available address for NVM write. */
+    uint32_t sectorSize; /*!< Size of erase sector. */
+    uint32_t totalSize; /*!< Total size of NVM storage. */
 } wsfNvmCb;
 
+extern uint32_t __pal_nvm_db_start__, __pal_nvm_db_end__;
+
 /**************************************************************************************************
   Global Functions
 **************************************************************************************************/
+/*************************************************************************************************/
+/*!
+ *  \brief  Get remaining space in WSF Allcoated NVM.
+ *  \return Bytes left.
+ */
+/*************************************************************************************************/
+int32_t WsfNvmGetRemainingSpace(void)
+{
+    const int32_t bytesLeft = wsfNvmCb.totalSize - (wsfNvmCb.availAddr - WSF_NVM_START_ADDR);
+    WSF_ASSERT(bytesLeft >= 0);
```

Comment:
I tested out the functionality and got caught up here. So would the recommendation be to disable asserts to prevent that, or should this just return false in that case?

Reply:
After running this code, restarting and running ```WsfNvmInit``` the available address is correct 

```    WsfNvmInit();

    uint8_t data[1000] = {0};
    bool_t writeOk = TRUE;
    uint32_t id = 1;

    while(writeOk)
    {
        writeOk = WsfNvmWriteData(id, data, sizeof(data), NULL);
        id++;
    }
```

```
(gdb) p wsfNvmCb
$37 = {
  availAddr = 16384,
  sectorSize = 8192,
  totalSize = 16384
}
```